src/components/Contacts/ContactDetails/ContactDetailsTab/ContactDetailsTab.test.tsx

```diff
@@ -129,8 +131,30 @@ describe('ContactDetailTab', () => {
     );
   });
 
+  it('should open edit contact modal', async () => {
+    const { queryByText, getAllByRole } = render(
+      <SnackbarProvider>
+        <TestRouter router={router}>
+          <MuiPickersUtilsProvider utils={LuxonUtils}>
+            <ThemeProvider theme={theme}>
+              <GqlMockedProvider<ContactDetailsTabQuery>>
+                <ContactDetailsTab
+                  accountListId={accountListId}
+                  contactId={contactId}
+                />
+              </GqlMockedProvider>
+            </ThemeProvider>
+          </MuiPickersUtilsProvider>
+        </TestRouter>
+      </SnackbarProvider>,
+    );
+    await waitFor(() => expect(queryByText('Loading')).not.toBeInTheDocument());
+    userEvent.click(getAllByRole('img', { name: 'Edit Icon' })[0]);
+    await waitFor(() => expect(queryByText('Edit People')).toBeInTheDocument());
+  });
+
   it('handles deleting contact', async () => {
-    const cache = new InMemoryCache({ addTypename: false });
+    const cache = new InMemoryCache();
```

Comment:
Note to self and others, if you are using fragments in your queries and or mutations, you have to have ```addTypename``` be ```true``` in order for the fragments to work correctly in the test. Forget the exact reason why. I read this in an article somewhere and remembered it when my data wasn't showing up in the test. I'll try to find the article/thread and link it here if I find it.

Reply:
I believe this was the article: https://levelup.gitconnected.com/gotchas-using-react-apollo-mockprovider-ec2a22a07e76